javawrapper/device/iothub-client-javawrapper/DeviceClient.java

```diff
@@ -0,0 +1,130 @@
+// Copyright(c) Microsoft.All rights reserved. 
+// Licensed under the MIT license.See LICENSE file in the project root for full license information. 
+
+package javaWrapper;
+
+import com.sun.jna.Pointer;
+
+import javaWrapper.Iothub_client_wrapperLibrary.AMQP_Protocol;
+import javaWrapper.Iothub_client_wrapperLibrary.HTTP_Protocol;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_HANDLE;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_RESULT;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_TRANSPORT_PROVIDER;
+import javaWrapper.Iothub_client_wrapperLibrary.IotHubEventCallback;
+import javaWrapper.Iothub_client_wrapperLibrary.IotHubMessageCallback;
+import javaWrapper.Iothub_client_wrapperLibrary.MQTT_Protocol;
+
+
+public final class DeviceClient 
+{
+	private static IOTHUB_CLIENT_HANDLE handle = new IOTHUB_CLIENT_HANDLE(null);
+	private IOTHUB_CLIENT_TRANSPORT_PROVIDER transport;
+	
+	protected DeviceClient()
+	{
+	
+	}
+	
+	public enum IotHubClientProtocol 
+	{ 
+		HTTPS, AMQPS, MQTT 
+	} 
+
+	public DeviceClient(String connString, IotHubClientProtocol protocol)
+	{
+		switch (protocol) 
+		{ 
+			case HTTPS: 
```

Comment:
Please add unit tests. For example, please follow the existing testing from the Java client. Feel free to reach out to Ilie with any questions. The c-code should be mocked and obviously not needed to be tested.